src/contrasts.jl

```diff
@@ -244,11 +244,11 @@ julia> StatsModels.ContrastsMatrix(StatsModels.FullDummyCoding(), ["a", "b", "c"
  0.0  0.0  0.0  1.0
 ```
 """
-type FullDummyCoding <: AbstractContrasts
+mutable struct FullDummyCoding <: AbstractContrasts
 # Dummy contrasts have no base level (since all levels produce a column)
 end
 
-ContrastsMatrix{T}(C::FullDummyCoding, levels::Vector{T}) =
+ContrastsMatrix(C::FullDummyCoding, levels::Vector{T}) where {T} =
```

Comment:
`T` isn't used at all?

Reply:
I wrote this code when I was just getting handle on type parameters, so I went a little overboard (better safe than sorry?)